infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -980,11 +896,10 @@ private AddressMetaData writeRecord(SegmentHandle segment, long address,
         ByteBuffer record = getByteBuffer(metadata, logEntry);
         long channelOffset;
 
-        try (MultiReadWriteLock.AutoCloseableLock ignored =
-                     segmentLocks.acquireWriteLock(segment.getSegment())) {
+        try (AutoCloseableLock ignored = segmentLocks.acquireWriteLock(segment.getSegment())) {
```

Comment:
You can remove `segmentLocks` it's not used, it will break one test in StreamLogFiles, but that use case isn't supported. We never have multi-writers

Reply:
wouldn't like to touch it in this PR, since it changes the logic a bit